### PR TITLE
refactor: use fbeta score to simplify metrics for classifiers

### PR DESCRIPTION
**Pull request by dvsrepo**
First version, still missing:

- Cleanup metric names to avoid tensorboard renaming during training.
- maybe simplify the get_metrics function with a helper function.
- @frascuchon @dcfidalgo I would like to make the per_label metrics configurable in classification heads, sometimes you don't want such level of detail and it makes very large logs, metric results files. But sometimes is very useful to have separate metrics for labels. I was thinking about a bool like `log_per_label_metrics` , what do you think?

### CODE REVIEW

**Comment by frascuchon** [comment]
> First version, still missing:
> 
> * Cleanup metric names to avoid tensorboard renaming during training.
> * maybe simplify the get_metrics function with a helper function.
> * @frascuchon @dcfidalgo I would like to make the per_label metrics configurable in classification heads, sometimes you don't want such level of detail and it makes very large logs, metric results files. But sometimes is very useful to have separate metrics for labels. I was thinking about a bool like `log_per_label_metrics` , what do you think?

+1 to configurable label metrics in training. For me is a sort of `verbose` training model more than a head configuration param.

**Comment by dcfidalgo** [comment]
> @frascuchon @dcfidalgo I would like to make the per_label metrics configurable in classification heads, sometimes you don't want such level of detail and it makes very large logs, metric results files. But sometimes is very useful to have separate metrics for labels. I was thinking about a bool like log_per_label_metrics , what do you think?

:+1: Maybe i would remove the "log_", since it suggests that they are calculated but not logged. Maybe just `metrics_per_label`?

**Comment by dvsrepo** [comment]
This is ready for review. Finally I have found by chance that allennnlp provides a mechanism for hidding metrics from being shown on progress bar (using underscore) so I have defined per_label metrics with underscore. In this way we keep things simple and get metrics per label always and only for tensorboard and summaries.

I have added clean_metric_name using the same code as tensorboardX, not sure if helpers is the best place for this, but might be useful in the future if we do other loggers (Mlflow, Weights and Biases, etc.)

File: src/biome/text/helpers.py [hunk 1]
```diff
@@ -88,3 +91,12 @@ def split_signature_params_by_predicate(
     non_matches_group = list(filter(lambda p: not predicate(p), parameters))
 
     return matches_group, non_matches_group
+
+
+def clean_metric_name(name):
+    if name is not None:
+        new_name = _INVALID_TAG_CHARACTERS.sub("_", name)
+        new_name = new_name.lstrip("/")
+        if new_name != name:
```
Comment (frascuchon) [inline src/biome/text/helpers.py:100]: this statement has no effect to final result

File: src/biome/text/modules/heads/classification/defs.py [hunk 2]
```diff
@@ -145,34 +150,18 @@ def get_metrics(self, reset: bool = False) -> Dict[str, float]:
...
-        all_metrics["average/precision"] = total_precision / num_classes
-        all_metrics["average/recall"] = total_recall / num_classes
-
-        return all_metrics
+        metrics = {}
+        for name, metric in self.metrics.items():
+            if name == 'accuracy':
+                metrics['accuracy'] = metric.get_metric(reset)
+            elif name in ['macro', 'micro']:
+                metrics.update({'{}/{}'.format(name, k): v for k, v in metric.get_metric(reset).items()})
```
[the remark this answers is not in the document]
Comment (dvsrepo) [reply src/biome/text/modules/heads/classification/defs.py]: done!

Forgot to add the underscore prefix :-)

File: src/biome/text/modules/heads/classification/defs.py [hunk 3]
```diff
@@ -145,34 +150,18 @@ def get_metrics(self, reset: bool = False) -> Dict[str, float]:
...
-                all_metrics[metric_name + "/recall"] = recall
-
-        num_classes = self.num_labels
-        all_metrics["average/f1"] = total_f1 / num_classes
-        all_metrics["average/precision"] = total_precision / num_classes
-        all_metrics["average/recall"] = total_recall / num_classes
-
-        return all_metrics
+        metrics = {}
+        for name, metric in self.metrics.items():
```
[the remark this answers is not in the document]
Comment (dvsrepo) [reply src/biome/text/modules/heads/classification/defs.py]: done!

File: src/biome/text/helpers.py (hunk 1 above)
Comment (dvsrepo) [reply src/biome/text/helpers.py]: done

**Pull request merged**